keep-ui/app/(keep)/alerts/EnrichAlertSidePanel.tsx

```diff
@@ -0,0 +1,239 @@
+        return;
+      }
+
+      const keys = customFields.map((field) => field.key);
+      const hasEmptyKeys = keys.some((key) => !key);
+      const hasDuplicateKeys = new Set(keys).size !== keys.length;
+
+      setIsDataValid(!hasEmptyKeys && !hasDuplicateKeys);
+    };
+
+    const calculateFinalData = () => {
+      return customFields.reduce(
+        (acc, field) => {
+          if (field.key) {
+            acc[field.key] = field.value;
+          }
+          return acc;
+        },
+        {} as Record<string, string>,
+      );
+    };
+    setFinalData(calculateFinalData());
+    validateData();
+  }, [customFields, preEnrichedFields]);
+
+  useEffect(() => {
+    if (!isOpen) {
+      setFinalData({});
+      setIsDataValid(false);
+    }
+  }, [isOpen]);
+
+  const handleSave = async () => {
+    const requestData = {
+      enrichments: finalData,
+      fingerprint: alert?.fingerprint,
+    };
+
+    const enrichedFieldKeys = customFields.map((field) => field.key);
+    const preEnrichedFieldKeys = preEnrichedFields.map((field) => field.key);
+
+    const unEnrichedFields = preEnrichedFieldKeys.filter((key) => {
+      if (!enrichedFieldKeys.includes(key)) {
+        return key;
+      }
+    });
+
+    let fieldsUnEnrichedSuccessfully = true;
+
+    if (unEnrichedFields.length != 0) {
+      const unEnrichmentResponse = await fetch(`${apiUrl}/alerts/unenrich`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${session?.accessToken}`,
+        },
+        body: JSON.stringify({
+          fingerprint: alert?.fingerprint,
+          enrichments: unEnrichedFields,
+        }),
+      });
+      fieldsUnEnrichedSuccessfully = unEnrichmentResponse.ok;
+    }
+
+    const response = await fetch(`${apiUrl}/alerts/enrich`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${session?.accessToken}`,
+      },
+      body: JSON.stringify(requestData),
+    });
+
+    if (response.ok && fieldsUnEnrichedSuccessfully) {
+      toast.success("Alert enriched successfully");
+      await mutate();
+      handleClose();
+    } else {
+      toast.error("Failed to enrich alert");
+    }
+  };
+
+  const renderCustomFields = () =>
+    customFields.map((field, index) => (
+      <div key={index} className="mb-4 flex items-center gap-2">
+        <TextInput
+          placeholder="Field Name"
+          value={field.key}
+          onChange={(e) => updateCustomField(index, "key", e.target.value)}
+          required
+          className="w-1/3"
+        />
+        <TextInput
+          placeholder="Field Value"
+          value={field.value}
+          onChange={(e) => updateCustomField(index, "value", e.target.value)}
+          className="w-full"
+        />
+        <Button color="red" onClick={() => removeCustomField(index)}>
+          ✕
+        </Button>
+      </div>
+    ));
+
+  return (
```

Comment:
You can use the side panel component you created, right?

Reply:
Yes, I'll add that once #2581 is merged.